database/memory_database_iterator.go

```diff
@@ -0,0 +1,80 @@
+package database
+
+import (
+	"github.com/syndtr/goleveldb/leveldb/util"
```

Comment:
try to avoid coupling memory database to level_db
you can declare a new iterator interface if it helps

Reply:
You're right, but I had to do it because database.DB expose leveldb iterator type.
Let me know if you have an idea how to tackle it

Reply:
just declare a releaser interface in that package and use it instead

Reply:
We are only using 2 functions for iteration, next() and release, you can create an interface containing only these functions. the levelDB iterator will implement these as well

Reply:
Thanks for the suggestions, made a new "base" iterator interface